### DISCUSSION TITLE
F-Strings

### DISCUSSION THREAD

**Discussion by froggestspirit** [godotengine/godot-proposals#4036] [Scripting]
I know this has been mentioned a little in the past, but it would be nice to see f-string support for the GDScript language.
https://docs.python.org/3/tutorial/inputoutput.html
I don't think it would need to be as in depth as python 3 (unless it works fine that way?), but something like
f"The value of x, is {x}" would be nice, and require less than format strings.

**Comment by ElliotBak3r** [comment]
Isn't that what the "%" character in strings is for ([Doc](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_format_string.html)) ?

**Comment by froggestspirit** [comment]
Yes, but it’s an alternative way to utilize it. Instead of
var script = "GDScript"
var name = "Estragon"
var format_string = "%s was reluctant to learn %s, but now he enjoys it."
var actual_string = format_string % [name, script]

it would be

var script = "GDScript"
var name = "Estragon"
var format_string = f"{name} was reluctant to learn {script}, but now he enjoys it."

**Reply by Yagich** [reply]
FWIW, the `String.format()` method has a similar syntax:
```gdscript
var script = "GDScript"
var name = "Estragon"
var format_string = "{name} was reluctant to learn {script}, but now he enjoys it."
var actual_string = format_string.format({"name": name, "script": script})
```
It's a handy function that accepts [plenty of different ways to format the string](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_format_string.html#format-method-examples).

**Comment by azur-wolve** [comment]
Its similar to C#: 
`$"string_literals {var_name} string_literals {obj.method()}";`
pretty handy

also for keep tabulation and new lines: 
`@"string_literals`
`string_literals`
`string_literals";`
(the new lines are preserved in the string)

Tbh feel the current f-strings in GDScript different than other languages (in a bad sense), it doesn't like me.

**Comment by dalexeev** [comment]
There is already a proposal:

* #157

Let's close this discussion for consolidation. Thanks for the contribution nonetheless!
